### ISSUE TITLE
Remaining Issues for WinFreeREG

### ISSUE THREAD

**Issue by SteveBiggs** [FreeUKGen/MyopicVicar#1422]
Collate any changes/fixes/enhancements required to get WinFreeREG ready for formal release.

**Comment by SteveBiggs** [comment]
Mike got WinFreeREG working with FR when it downloaded Place and Church names for selection. Kirk had set up a special screen with all the information needed. But as FR developed this screen was changed and Mike never got WinFreeREG to interface with it. So that needs fixing first because without the list of Places and Churches the rest gives errors when someone starts a new batch.

**Comment by SteveBiggs** [comment]
I am still seeing unhandled exception errors - this one always occurs when I select the "File Details" icon:
![Capture.GIF](https://waffleio-direct-uploads-production.s3.amazonaws.com/uploads/588631f51e19621a00230146/125516c66e82c728ace21e0d46cb958a379791accda5fd53b652fa912a5265296b51f02e7384e671e6162351580d4bac43585f45ffac8928f0a42062c80d55b98e30063e937f03a6aa8318b500bf228a4d85b1bd8d4ac8ed3937aa.GIF)

**Comment by SteveBiggs** [comment]
Needs to account for flexible csv functionality.

**Comment by edickens** [comment]
I think flexible csv needs to be a new story so that we get WinFreeREG working first before we do more development.  E

**Comment by SteveBiggs** [comment]
Perhaps, but flexible csv is already implemented in FR and Kirk is adding the extra fields now, so we need it supported in WinFreeREG as soon as possible.

**Comment by SteveBiggs** [comment]
At the end of a line, hitting 'Enter' opens the next line but you have to click in a box to start editing. Can it be made to work like WinREG so that 'Enter' opens the first field in the new line to start typing immediately?

**Comment by EmJayFry** [comment]
Since I can't attend the scrums, I'm constrained to communicating either this way or (preferably)  by email.

Current status of WinFreeREG, as of today, is as follows

1) There are presently two screens for entering data. Firstly, the original method of using a somewhat clunky grid. This has had a lot of tweaking and is relatively usable. Secondly, I have added the same datagrid as is used in WinREG. This has been simplified and is very usable. 

I am currently in the process of adding a third input screen. This will replicate the alternative screen found in WinREG where a record is selected from a list and edited field by field.

The above will be available as version 1.1.11. Version 1.1.10 is available for download upon request to me. Ali already has it and seems quite impressed with it.

Once I've got 1.1.11 working, the next step will be to make a start on this so-called flexible-csv. Initially, all I want to do is to create all new files with the flexible-csv headers for ONLY the standard fields. I then want to be able to edit these new files. Last stage could be to convert existing files to use the flexible-csv headers when they are edited. This to be optional for the user. This will be version 1.1.12.

Finally, we need to devise a means of dropping some field headings from a file and also of adding new fields to a file. Version 1.1.14?

**Comment by Captainkirkdawson** [comment]
Mike good to see that communication. This is definitely the referred  way of communicating on technical matters. 
Could you comment on the status with respect to the servers and uploading?

**Comment by EmJayFry** [comment]
I'm not sure what you want to know. Are you referring to whether WinFreeREG can upload or not - it can, or used to be able to using servers test3 and live.

⁣-- 
Regards
Mike Fry (Johannesburg)

**Comment by Captainkirkdawson** [comment]
That is exactly the question. Can it be used to load files into the production servers and also test3.

**Comment by EmJayFry** [comment]
Well, as I said, it was working very well last year, and so long as you haven't changed much in the interim I see no reason why it shouldn't still work. I can do nothing now until Monday. If it's urgent you could ask Ali Hill.

⁣-- 
Regards
Mike Fry (Johannesburg)

**Comment by SteveBiggs** [comment]
I've just successfully uploaded a file to production, so can confirm it is now working.

Also fetching of churches works fine, but when I try to fetch places I get an unhandled exception:

![image](https://user-images.githubusercontent.com/25303607/37532652-818a4f8c-2937-11e8-9c48-532ae150a029.png)

**Comment by EmJayFry** [comment]
Leave me an email detailing what steps you followed to cause the problem along with whatever info you can copy and paste from under the Details button. I'll have a look on Monday once I can get to my machine again.

⁣-- 
Regards
Mike Fry (Johannesburg)

**Comment by EmJayFry** [comment]
Looking at the error again, I think I've seen it before. If so, I think I've already fixed it in version 1.1.10

⁣-- 
Regards
Mike Fry (Johannesburg)

**Comment by SteveBiggs** [comment]
I have managed to locate the 1.1.10 download site from previous emails and installed that. So far it looks as though it does not have any of the unhandled exception problems.

**Comment by EmJayFry** [comment]
Just this minute got message from Ali Hill and the url should be

http://stash.freeukgen.org.uk/winreg/WinFrwrteeReg.1.1.10.msi
⁣-- 
Regards
Mike Fry (Johannesburg)

**Comment by SteveBiggs** [comment]
So my only outstanding request is:
At the end of a line, hitting 'Enter' opens the next line but you have to click in a box to start editing. Can it be made to work like WinREG so that 'Enter' opens the first field in the new line to start typing immediately?

**Comment by EmJayFry** [comment]
I've tried and tried to get this to work in a fluid manner but failed with every attempt. Hence the desire to add identical input screens to those found in WinREG. The datagrid from there does exactly what you want.

⁣-- 
Regards
Mike Fry (Johannesburg)

**Comment by SteveBiggs** [comment]
Not quite. The data grid works using 'tab' to next line first field, but not 'enter' (which goes to the field directly below where you are). But I guess you've done all you can to try to make that work.

**Comment by EmJayFry** [comment]
And this is in WinREG?

⁣-- 
Regards
Mike Fry (Johannesburg)

**Comment by SteveBiggs** [comment]
In WinREG, hitting 'enter' moves you to the next field along a line, when you are at the end it moves you to the first field in the next line.

In WinFreeREG it takes you down to the field below the one you are on.

**Comment by EmJayFry** [comment]
Ah! But you've yet to experience the WinREG grid in its WinFreeREG guise ;-)

⁣-- 
Regards
Mike Fry (Johannesburg)

**Comment by SteveBiggs** [comment]
I am talking about the grid version.

**Comment by EmJayFry** [comment]
Without access to my machine and the code I'd be stupid to try and comment further with any sort of authority. Leave it till Monday and I'll see what the story really is.

⁣-- 
Regards
Mike Fry (Johannesburg)

Sent from BlueMail ​On 16 Mar 2018, at 20:28, Stephen Biggs <notifications@github.com> wrote:

I am talking about the grid version.

—
You are receiving this because you were assigned.
Reply to this email directly, view it on GitHub, or mute the thread.

**Comment by SteveBiggs** [comment]
Further use of the gridview in WinFreeREG shows that the names are not auto-corrected for upper/lower case as they are in the Data list view.

**Comment by Captainkirkdawson** [comment]
@EmJayFry  wrote by email
I have, today, made a preliminary issue of version 1.1.11 available from the usual place. If you're currently using 1.1.10, then the Assistance menu can take you to the home website from where the new version can be downloaded.

In case you're not the url is

http://stash.freeukgen.org.uk/winreg/WinFreeReg.1.1.11.msi

The first thing to note is that I have grafted most of the WinREG look and feel onto the program so our more conservative users should feel more comfortable with the program.

I think there are one or two rough areas that will need to be sorted out. I'm relying on your feedback to point me towards where further work is necessary.

**Comment by Captainkirkdawson** [comment]
@SteveBiggs  responded

Personally, I'm happy with the look of either the Data List or Grid views. My only comment is the one I've made before which is that it would be nice if when you get to the end of a line and hit 'enter', it worked like WinREG in opening the next line and placing you in the FIRST field of that new line, so you can smoothly continue transcribing without needing to click in a field at the start of each line.

Steve

**Comment by Captainkirkdawson** [comment]
Mike said

Just released version 1.1.12 and uploaded it to the usual place,

I think Steve will be pleased with the way the ENTER key now behaves on the WinREG-lookalike grid :-) It moves from field to field, record to record, left to right and down the file. At the end of the file a new record gets added automatically.

Just prior to releasing this, I found some problems with being able to Show and Hide columns on one display and have the effects reflected on the other two displays. This has been changed but doesn't work as I intended it to. There may be some residual problems.

I have also changed the Relationship and Condition columns in the datagrid from using ComboBoxes to being just plain, ordinary TextBoxes. I did this because the ComboBox besides restricting the user to using what is in the relevant table, didn't display anything that wasn't in the table. An error was indicated but the erroneous data didn't show. With a TextBox this doesn't happen. The errors are still noted and the invalid data now shows. Unfortunately, the change has temporarily removed the AutoCompletion ability that was present with the ComboBoxes.

**Comment by SteveBiggs** [comment]
Some observations on testing using 1.1.12:
1. When a new file is started, the file details box stays open behind the grid view obscuring the image you're transcribing  from, so you have to close the file then re-open it.
2. The Abode doesn't auto-capitalise.
3. When you hit enter, you can't see which field is active.
4. On every other line when hitting enter, a new line opens but typing stays on current line so you have to click the first field of the next line.

**Comment by EmJayFry** [comment]
Notes. Will do something next week. Auto-capitalism of Abode could be problematic. C.f. Cley next the Sea

⁣-- 
Regards
Mike Fry (Johannesburg)

**Comment by SteveBiggs** [comment]
Thanks Mike. I think we can only expect the first letter of each sentence to be auto-cap. In fact it would be good if this was the case for all free-text fields (profession, notes, etc.) as was the case in WinREG. If there are other odd capitalisation requirements, the transcriber will need to edit accordingly.

**Comment by edickens** [comment]
I recommend no auto-cap.  It is not essential.  And what about entries such as "de Vere".

**Comment by Sherlock21** [comment]
I agree with Eric D

Its likely to introduce as many errors as it corrects, as pointed out in the previous post.

**Comment by SteveBiggs** [comment]
If "de Vere" is a surname it will be fully capitalised to DE VERE like all surnames.

When you are transcribing, the less you have to think about capitalising, the better. The great majority of fields need a capital for the first letter or immediatley after a full stop. If they don't, the transcriber can edit accordingly but it flows much better with the default being capitalise first letters.

**Comment by Sherlock21** [comment]
If de Vere is a Surname of one of the key players, then yes all gets capitalised.

BUT if de Vere is any other field - either as a first name or as a surname, then NO.  It should remain as de Vere. So its best left up to the Transcriber to get it right. We have been round this loop many years ago, and the conclusive decision was to NOT have the program - either FR's code Nor WinReg  to change. 

The logic has not changed in my view.

**Comment by edickens** [comment]
I only chose "de Vere" as something starting with lowercase.  I can't think of a place at present, but I am sure there are some.  I know that lastnames are all capital.  E

**Comment by Sherlock21** [comment]
There are many Place names that contain elements in lower case.

A few selected at random:
Ashby by Partney 
Ashby cum Fenby 
Ashby de la Launde
all in LIN.
and in DBY:
Alsop en le Dale

and onwards. there are quite a lot.

**Comment by SteveBiggs** [comment]
Yes, and the transcriber must have the ability to edit accordingly. My only point is that for 99% of the time, the first letter of the abode (and all other fields) is a capital letter and it saves a little transcribing time if the default is that WinFreeREG makes it a capital. In the few cases where it's not, the transcriber must be allowed to change it though.

**Comment by SteveBiggs** [comment]
All of the places Eric lists do start with a capital letter.

**Comment by EmJayFry** [comment]
If you give me a list of 'prefixes' then I can make sure that surnames like de Vere end up on FR as 'de VERE' rather than as 'DE VERE' :-) On the whole I think I prefer the latter.

Abodes, parishes and occupations can all be treated the same: capitalisation of the initial letter.

Does that also apply to Notes? Here there can be proper sentences, so the first letter of each sentence can be capitalised. Of course, a sentence needn't necessarily end with a full stop.

⁣-- 
Regards
Mike Fry (Johannesburg)

**Comment by SteveBiggs** [comment]
Thanks Mike. A key point is that although the default is capitalisation of the initial letter and first letter of each sentence, it must be possible for the transcriber to override that and change to lower case if applicable.

**Comment by Sherlock21** [comment]
Exactly. Thanks for clarifying that. I had assumed that it was proposed to make WinReg's version not editable - except by swapping over to Spreadsheet perhaps.

**Comment by Sherlock21** [comment]
re Mike's point about Case in NOTES fields.

I would be strongly against changing any Case in notes - regardless of what you decide for names.
Reason: I have written many notes, where say a name is in the middle of a sentence - like but not restricted, to Additional Witnesses. Or someones Qualifications. or an expansion of their Abode. I am sure there re WinReg users who have done the same. SO I think thats dangerous - even if you make it subsequently editable in WinReg, back to what the user typed in the first place.

**Comment by SteveBiggs** [comment]
This is OBE as WinFreeREG is dead but the discussion may be of use when considering requirements for FROTT?
